spyder/app/mainwindow.py

```diff
@@ -1448,17 +1448,27 @@ def post_visible_setup(self):
         self.menuBar().raise_()
         self.is_setting_up = False
 
-        # Handle DPI scale changes and show restart message
-        screen = self.window().windowHandle().screen()
-        screen.logicalDotsPerInchChanged.connect(self.show_DPI_change_message)
+        # Handle DPI scale and window changes to show a restart message
+        window = self.window().windowHandle()
+        window.screenChanged.connect(self.handle_new_screen)
+        self.screen = self.window().windowHandle().screen()
+        self.screen.logicalDotsPerInchChanged.connect(
+            self.show_DPI_change_message)
 
         # Notify that the setup of the mainwindow was finished
         self.sig_setup_finished.emit()
 
+    def handle_new_screen(self, screen):
+        """Connect DPI signals for new screen."""
+        self.screen.logicalDotsPerInchChanged.disconnect(
+            self.show_DPI_change_message)
+        self.screen = screen
+        self.screen.logicalDotsPerInchChanged.connect(
+            self.show_DPI_change_message)
+
     def show_DPI_change_message(self):

```

Comment:
Let's change the name of this method to `show_dpi_change_message`. It's a bit odd to have capitalized letters in the middle of a method name.